### PR TITLE
Fix #611: Clamp maxDistance and refDistance

### LINKED ISSUE #611: Distance models: maxDistance vs refDistance
What should happen if someone sets maxDistance < refDistance?

What should happen if someone sets refDistance = 0?  

Since these are all distances, presumably, we should throw an InvalidStateError if they are set to negative values.

### PR DESCRIPTION
**Pull request by rtoy**
For the linear distance model, make d'max = max(dref, dmax) and d'ref
= min(dref,dmax).  And specify the value of the model is 1-f if
d'ref=d'max. This makes the model well-defined for all values of dref
and dmax.

Note that this change allows a user to set dref > dmax, but the
behavior is well-defined in this case (by reversing the order). Not
sure if we want that or not.

Also, specify that setting refDistance and maxDistance to a
non-positive value throws a TypeError.

### CODE REVIEW

**Comment by rtoy** [comment]
PTAL. 

Note that the change allows dref > dmax, but the linear model is still well-defined in this case and also the case where dref = dmax (having the value 1-f).  Not sure we want that.

I also took the opportunity to say that maxDistance and refDistance must be strictly positive values.

File: index.html [hunk 1]
```diff
@@ -6396,7 +6401,9 @@ <h2 id="AudioWorklet-Examples">
           <dd>
             <p>
               A reference distance for reducing volume as source moves further
-              from the listener. The default value is 1.
+              from the listener. The default value is 1. A
+              <code>TypeError</code> exception must be thrown if this is set to
+              a non-positive value.
```
Comment (padenot) [inline index.html]: This is not a type issue, this is a range issue.

File: index.html [hunk 2]
```diff
@@ -6406,7 +6413,8 @@ <h2 id="AudioWorklet-Examples">
             <p>
               The maximum distance between source and listener, after which the
               volume will not be reduced any further. The default value is
-              10000.
+              10000. A <code>TypeError</code> exception must be thrown if this
```
Comment (padenot) [inline index.html]: Same.

File: index.html (hunk 1 above)
Comment (rtoy) [inline index.html]: RangeError then?

Comment (padenot) [inline index.html]: Usually "IndexSizeError" for that kind of things, although it's weird. RangeError does not exist any more apparently.

Comment (rtoy) [inline index.html]: Really?  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError still lists RangeError and https://heycam.github.io/webidl/#exceptiondef-rangeerror still lists it too.

**Comment by padenot** [comment]
You're right, I got something mixed up here.

**Pull request merged**